api/src/nlp/controllers/nlp-sample.controller.ts

```diff
@@ -294,6 +302,14 @@ export class NlpSampleController extends BaseController<
     @Body() { entities, language: languageCode, ...sampleAttrs }: NlpSampleDto,
   ): Promise<NlpSampleFull> {
     const language = await this.languageService.getLanguageByCode(languageCode);
+
+    if (!language) {
+      this.logger.warn(`Unable to Language by languageCode ${languageCode}`);
+      throw new NotFoundException(
+        `Language with languageCode ${languageCode} not found`,
+      );
+    }
```

Comment:
How about we put this check inside getLanguageByCode() and ensure that it always returns a language ?

Reply:
Yes i will implement the change